Instant Search: Update styles to reduce usage of "!important".

a8ck3n opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to projects/plugins/jetpack/to-test.md in a new commit as part of your PR. -->
<!-- a12s: If you have an expected version that you're aiming for the PR to add, please use the Milestone field to communicate it. If you leave it blank, that indicates there isn't a preference. -->

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->

Introduces some more specific CSS rules for the various styled buttons in the Overlay. Separated out and limited to those styles (font/color attributes) that are currently dependant on `!important` directives. Necessary to prevent aggressive theme styles from bleeding into the Overlay.

With this technique, it would still be possible for site owners to override the styles as needed.

Also updates the font weights to match what we had initially before the Twenty Twenty One issues surfaced.

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

No

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

1. Visit a site with a Jetpack Search plan.
2. Make sure the Twenty Twenty One theme is active. It has the aggressive theme styles we are trying to address.
3. Open the Overlay by appending `?s=&category=uncategorized` to your site address.
4. Confirm the Sort Options buttons look correct.
5. Confirm the "Clear filters" button looks correct.
6. Clear the filters and confirm the "Load more" button looks correct.
7. Inspect the various buttons and confirm there are no `!important` attributes used for size/color.

You can toggle the light/dark themes from `/wp-admin/admin.php?page=jetpack-search-configure`.

a8ck3n committed:
Instant Search: Fewer “!important” styles.

The various styled buttons in the Overlay now use a more specific set of CSS rules when setting font/color attributes. Necessary to prevent aggressive theme styles from bleeding into the Overlay.

With this technique, it would still be possible for site owners to override the styles as needed.

kangzj commented on projects/packages/search/src/instant-search/lib/styles/_mixins.scss:
```diff
@@ -109,14 +109,14 @@ $customberg-container-selector: '.jp-search-configure-app-wrapper' !default;
-	color: inherit !important;
+@mixin apply-default-overlay-colors() {
+	background-color: inherit;
+	color: #646970;
```
The color look quite pale on dark theme,

<img width="209" alt="image" src="https://user-images.githubusercontent.com/1425433/186823169-a2d82ec0-3ced-4384-b96a-599421526880.png">

Compared to trunk:

<img width="198" alt="image" src="https://user-images.githubusercontent.com/1425433/186823229-6cad3395-5c9b-43ac-8165-42e6e54e4ddc.png">

Could we use different colors for dark/light theme?

a8ck3n replied:
Recent changes to fix the button styling did not include the font colors/weights we originally had. I didn't see any comment that this was intentional though, so this change reintroduces the values we've had all along. The white in your screenshot has only been a part of the styling for about 2 weeks. Prior to that, it looked like it does in this branch.

That said, we could definitely go with white (or close to it) but we'd be breaking from the predefined theme colors. I'll have another look at this and see what I can come up with.

a8ck3n replied:
After some more digging, I see we do have colours defined for these controls in our dark mode theme. They were just being overridden the same way the light colours were. I've added extra specificity to those rules so they should take effect correctly now. I'll add a screenshot to show the updated appearance.

kangzj commented on projects/packages/search/src/instant-search/components/search-filters.scss:
```diff
@@ -19,8 +19,6 @@
 .jetpack-instant-search__clear-filters-link {
-	@include inherit-parent-colors-important();
-	@include inherit-parent-font-size-important();
 	border: none;
```
Is there any particular reason we didn't put these in the same selector on L36?

a8ck3n replied:
I'm not sure about that. Those changes were made while I was away so I wasn't part of any discussions that might have happened there. It was a bit of a rush to get the visual fixes into an Atomic update, I think.

That said, I could see an argument for keeping font metrics and colours as separate mixins.

kangzj reviewed (commented):
The overridden works as described. Left a couple of comments 😄

a8ck3n committed:
Update CSS for sort options in dark mode.

These were present but overridden in some themes. Adding extra specifity to rules so they are more likely to take effect.

a8ck3n commented:
Hey @kangzj. Thanks for the review and comments!

I've got some changes in that enable the existing dark theme colours to work again. Per the original issue, this was a matter of overly aggressive rules in some themes overriding our styles. Some extra specificity in our existing dark theme CSS allows our colours to properly take effect.

![SortOptions](https://user-images.githubusercontent.com/40267301/187398558-af37cd4f-bf25-4805-85a8-f8ce5854bf05.png)

a8ck3n committed:
Merge branch 'trunk' into fix/search-overlay-update-important-styles-2

a8ck3n committed:
Fix-up versions.

a8ck3n commented:
Merged changes from trunk, fixed up versions, and now passing CI tests. Ready for another review, please!

oskosk commented:
This looks good.

Tested on JN using Twenty Twenty one... These seem to be the divergences between trunk and this branch. What's highlighted in green is what I see that differs from trunk

**Twenty Twenty one  running this branch**

![image](https://user-images.githubusercontent.com/746152/187677875-5832f10e-e3f2-49b6-9ee1-1c16f0070818.png)

**_Load more_**

![image](https://user-images.githubusercontent.com/746152/187680192-9c758a30-fa75-434a-ad6d-d2d8f5c8f31b.png)



**Twenty Twenty one  running Trunk**

<img width="1501" alt="image" src="https://user-images.githubusercontent.com/746152/187678051-6f3d87ed-bc25-4583-aba9-66a97215b4ed.png">


**_Load more_**

<img width="285" alt="image" src="https://user-images.githubusercontent.com/746152/187677495-58ea2dcf-f833-403f-b399-b0b5a208ca6c.png">

a8ck3n commented:
Thanks for the review/merge @oskosk !
